### ISSUE TITLE
Write resource management overview

### ISSUE THREAD

**Issue by bgrant0607** [kubernetes/kubernetes#15673]
We need an overall design for how all our resource-management features should interact:

* horizontal pod autoscaling
* vertical pod autoscaling (initial and reactive)
* resource metrics APIs
* oversubscription
* scheduling (including taking resource usage into account)
* QoS
* out-of-resource killing
* rescheduling

cc @davidopp @fgrzadkowski @dchen1107 @vishh

**Comment by dchen1107** [comment]
+1000 

I can contribute some for sure.

**Comment by davidopp** [comment]
I agree. Producing such a doc was one of the goals of the meeting we had in Warsaw a couple of months ago, but that never happened. Probably all for the best, as we've made good progress in a number of these areas since then, so if we do it now, it will be more grounded in reality.

I'll assign this to myself for now, but if someone wants to start working on it before I get a chance, they should feel free.

**Comment by hurf** [comment]
@hurf @HaiyangDING

**Comment by bgrant0607** [comment]
OOR proposal: #17186 

Node resource management proposals: #14943, #13006, #16889

**Comment by dchen1107** [comment]
The initial issue on memory management: https://github.com/kubernetes/kubernetes/issues/14532
Node Allocatable issue: https://github.com/kubernetes/kubernetes/issues/13984 and proposal: https://github.com/kubernetes/kubernetes/pull/17201

**Comment by feiskyer** [comment]
+1

**Comment by mwielgus** [comment]
No traffic on the issue for more than a year. Can it be closed?

**Comment by vishh** [comment]
I think this is still necessary. @davidopp and I can own this.

**Comment by bgrant0607** [comment]
I mentioned this to @bsalamat last week

/remove-lifecycle stale

**Comment by resouer** [comment]
I would also like to help, it's really useful. Hope to see any lead of scheduling/resource management can own this work.

**Comment by thockin** [comment]
No sure what to do with this issue - yes having a holistic design doc would be good, but this issues isn't going to make that happen.  Is someone going to try to write it?
